api/raft.cc

```diff
@@ -56,7 +56,7 @@ void set_raft(http_context&, httpd::routes& r, sharded<service::raft_group_regis
         });
 
         if (!found_srv) {
-            throw std::runtime_error{fmt::format("Server for group ID {} not found", gid)};
+            throw bad_param_exception{fmt::format("Server for group ID {} not found", gid)};
```

Comment:
This is an unrelated change. It's nice, because you want to make the APIs consistent -- but please put it in a separate commit to make it clear that it's not part of implementing read barrier.

Reply:
(The commit should probably come before this one, so in this one you can use `bad_param_exception` without introducing inconsistencies)

Reply:
moved to a separate commit